test/libsolidity/syntaxTests/types/var_type_suggest.sol

```diff
@@ -0,0 +1,23 @@
+contract C {
+    function h() internal pure returns (uint, uint, uint) {
+        return (1, 2, 4);
+    }
+    function g(uint x) internal pure returns (uint) {
+        return x;
+    }
+    function f() internal pure {
+        var i = 31415;
+        var t = "string";
+        var g2 = g;
+        var myblockhash = block.blockhash;
+        var (a, b) = (2, "troi");
+        var (x,, z) = h();
+    }
+}
+// ----
+// SyntaxError: (224-237): Use of the "var" keyword is disallowed. Use explicit declaration `uint16 i = ...´ instead.
+// SyntaxError: (247-263): Use of the "var" keyword is disallowed. Use explicit declaration `string t = ...´ instead.
+// SyntaxError: (273-283): Use of the "var" keyword is disallowed. Use explicit declaration `function (uint256) pure returns (uint256) g2 = ...´ instead.
```

Comment:
Did you check if that works? Doesn't it have to be `function (uint256) pure *internal* returns ...'?